### PR TITLE
SSCS-10357 add appellants name when generating notice

### PR DESCRIPTION
**Pull request by pk2453**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionServiceBase.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ protected NoticeIssuedTemplateBody createPayload(PreSubmitCallbackResponse<SscsC
             writeFinalDecisionBuilder.dateOfDecision(null);
         }
 
-        writeFinalDecisionBuilder.appellantName(buildName(caseData, true));
+        writeFinalDecisionBuilder.appellantName(buildName(caseData, false));
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionServiceBase.java:105]: Isn't this just going to add the appointee name?

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionServiceBase.java (hunk 1 above)
Comment (pk2453) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionServiceBase.java:105]: > Isn't this just going to add the appointee name?

    protected String buildName(SscsCaseData caseData, boolean displayAppointeeName) {
        if (displayAppointeeName && "yes".equalsIgnoreCase(caseData.getAppeal().getAppellant().getIsAppointee())
            && null != caseData.getAppeal().getAppellant().getAppointee()) {
            return WordUtils.capitalizeFully(caseData.getAppeal().getAppellant().getAppointee().getName()
                    .getFullNameNoTitle(), ' ', '.');
        }
        return WordUtils.capitalizeFully(caseData.getAppeal().getAppellant().getName()
                .getFullNameNoTitle(), ' ', '.');
    }

true -> appointee name
false -> appellant name

Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionServiceBase.java:105]: Don't know why I read that backwards, is there a test for this change?

Comment (pk2453) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionServiceBase.java:105]: there is already test for **buildName** which covers both scenario.

Comment (pk2453) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionServiceBase.java:105]: additional check in test added

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by ikirsanov — APPROVED** [review]

**Comment by filippo-possenti-version1** [comment]
Superseded by https://github.com/hmcts/sscs-tribunals-case-api/pull/2953 which was opened to overcome the difficulties encountered during some merge operations on this one.

**Pull request closed without merging**